src/core/pointcloud/qgseptdecoder.cpp

```diff
@@ -579,38 +617,49 @@ QgsPointCloudBlock *__decompressLaz( FileType &file, const QgsPointCloudAttribut
 
       outputOffset += requestedAttribute.size;
     }
+
+    // check if point needs to be filtered out
+    if ( filterExpression.isValid() )
```

Comment:
again, determine once upfront out of the point loop

Reply:
ok